sklego/model_selection.py

```diff
@@ -80,29 +95,61 @@ def split(self, X, y=None, groups=None):
 
         date_min = X_index_df["__date__"].min()
         date_max = X_index_df["__date__"].max()
+        date_length = X_index_df["__date__"].max() - \
+            X_index_df["__date__"].min()
+
+        if ((self.train_duration is None)
+                and (self.n_splits is not None)):
+            self.train_duration = date_length - \
+                (self.gap_duration + self.valid_duration * self.n_splits)
+
+        if ((self.train_duration is not None)
```

Comment:
This check already happens in the `__init__` right? Is it needed to do it again here/

Reply:
Yes it does, train_duration can be None and thus the check only happens partly in the init, if train duration is calculated using valid_duration and n_splits, I do the check again